linux/utils/collect_dmesg.yml

```diff
@@ -4,25 +4,43 @@
 # Description:
 #   Collect dmesg and save output to a local file
 # Parameters:
-#   dmesg_opts: The dmesg command options
-#   dmesg_output_file_src: Save dmesg output file on guest OS
+#   dmesg_options: the dmesg command options
+#   dmesg_output_file_name (optional): the file name for saving dmesg output in test case log folder
+#   dmesg_ignore_errors (optional): ignore errors when executing dmesg. Default is false.
 # Return:
-#   dmesg_output_file_local: The dmesg output file at localhost
+#   dmesg_output_file_path: the file path to saved dmesg output at localhost
 #
-- name: "Print dmesg ring buffer to {{ dmesg_output_file_src }}"
-  ansible.builtin.shell: "dmesg {{ dmesg_opts | default('') }} >{{ dmesg_output_file_src }}"
+- name: "Initialize facts of dmesg command result and output file path at localhost"
+  ansible.builtin.set_fact:
+    dmesg_cmd_result: ""
+    dmesg_output_file_path: ""
+
+- name: "Print dmesg ring buffer"
+  ansible.builtin.shell: "dmesg {{ dmesg_options | default('') }}"
   changed_when: false
+  ignore_errors: "{{ dmesg_ignore_errors | default(false) }}"
   delegate_to: "{{ vm_guest_ip }}"
+  register: dmesg_cmd_result
 
-- name: "Fetch dmesg output to test case log folder"
-  include_tasks: fetch_file.yml
-  vars:
-    fetch_file_src_path: "{{ dmesg_output_file_src }}"
-    fetch_file_dst_path: "{{ current_test_log_folder }}/"
-
-- name: "Set facts of dmesg output file at localhost"
-  ansible.builtin.set_fact:
-    dmesg_output_file_local: "{{ fetch_file_local_path }}"
+- name: "Save dmesg output to local file in test case log folder"
   when:
-    - fetch_file_local_path is defined
-    - fetch_file_local_path
+    - dmesg_output_file_name is defined
```

Comment:
So this parameter 'dmesg_output_file_name' does not have a default value, if it's not defined, then will not save output to a file.